code/__HELPERS/unsorted.dm

```diff
@@ -1458,3 +1458,9 @@ GLOBAL_DATUM_INIT(dview_mob, /mob/dview, new)
 			else
 				return "\[[url_encode(thing.tag)]\]"
 	return "\ref[input]"
+
+// Checks if the turfs two atoms are on are within range of each other.
+/proc/in_range(atom/source, atom/user)
+	var/turf/T1 = get_turf(source)
+	var/turf/T2 = get_turf(user)
+	return T1 && T2 && T1.z == T2.z && get_dist(T1, T2) <= 1
```

Comment:
Just make sure every use of in_range() uses typecasted args and turn this back into a define

Reply:
Almost every use of `in_range` is being called on objs or mobs, this will mean up to 250 new lines of `var/turf/foo_turf = get_turf(foo)`.

Reply:
I'm unsure if that's worth it. Probably but let another maintainer chime in on that.

Reply:
```#define in_range(source, user) (get_dist(source, user) <= 1 && (get_turf(source):z) == (get_turf(user):z))```
Only works in 512, but it does work.

Reply:
we should go full 512 anyways

Reply:
By the time we have any multiz stuff we'll be 512 anyway, I say wait until we hit 512